index.js

```diff
@@ -449,6 +450,17 @@ module.exports = {
 						}
 						return ctx.succeed(result);
 					};
+
+					// adds values from withSessionAttributes to the session
+					if (currentItem.withSessionAttributes) {
+						var session = request.session.attributes;
+
+						for (var newAttribute in currentItem.withSessionAttributes) {
+							if (typeof session[`${newAttribute}`] === "undefined") {
```

Comment:
Why not `if (!session[newAttribute]) {`

Reply:
Good suggestion. I've had issues before with `"undefined"` vs `undefined`, but that syntax should clear things up.